### PR TITLE
Implement our own interfering process killer

### LINKED ISSUE #643: Wifiphisher stoped my networkmanager, even after reboot
Downloaded latest wifiphisher with dbus. When started network-manager vas closed directly on first start.
But when i closed the tool and type as usual "service network-manager start" nothing happend. 
Rebooted my computer and still no wifi icon in upper right corner.
Whats the command to let dbus start networkmanager again?

### PR DESCRIPTION
**Pull request by anakin1028**
@sophron @blackHatMonkey  I think this should fix #643

### CODE REVIEW

File: wifiphisher/pywifiphisher.py [hunk 1]
```diff
@@ -494,25 +495,30 @@ def display_access_points(information, mac_matcher):
...
+    output = proc.communicate()[0]
+    # total processes in the system
+    sys_procs = output.splitlines()
+    # loop each interfering processes and find if it is running
+    for interfering_proc in INTEFERING_PROCS:
+        for proc in sys_procs:
+            # kill all the processes name equal to interfering_proc
+            if interfering_proc in proc:
+                pid = int(proc.split(None, 1)[0])
+                print '[' + R + '!' + W + '] Kill ' + interfering_proc
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: Can we change this to: "[+] Sending SIGKILL to " + interfering_proc?

Comment (sophron) [inline wifiphisher/pywifiphisher.py]: (The plus sign in green.)

**Comment by sophron** [comment]
Looks good. As soon as you fix the `print` I will merge this.

File: wifiphisher/pywifiphisher.py (hunk 1 above)
Comment (anakin1028) [inline wifiphisher/pywifiphisher.py]: fixed that : )

**Comment by sophron** [comment]
Thanks!

**Comment by squashen** [comment]
Have downloaded and tryed.
The tool started and closed network-manager, but dont start it after closing. 
But this time i didnt need to "nmcli n on" it, only i had to do was " /etc/init.d/network-manager restart" and icon showed up.

**Comment by anakin1028** [comment]
Thanks @squashen. Can you observe that what processes the tool he killed?  I remembered when you commented out two lines of the code, the icon will automatically show up right? I guess we may kill the wpa_supplicant belonging to networkmanager which cause it not automatically restart

**Comment by squashen** [comment]
Strange... he "Sending SIGKILL to wpa_supplicant and Sending SIGKILL to dhclient"
But this time i had to do the nmcli n on before restart network-manager:S

**Comment by squashen** [comment]
Hmm ahh, this time i took a shortcut and pressed ctrl-c just when he killed the process, if i wait and chose network and attacktype it is enough with just " /etc/init.d/network-manager restart"

**Comment by anakin1028** [comment]
@squashen yes, this a reasonable observation. Since when you pressed ctrl-c just after the processes are killing, we don't catch exception that time. If you wait until the main terminal, we'll do the NetworkManager recover.

**Comment by anakin1028** [comment]
if you are interested in this issue, you may try to remove the processes `wpa_supplicant` and `dhclinet` in the `constant.py` file and see if the NetworkManager automatically restart.

```python
INTEFERING_PROCS = ["wpa_action", "wpa_supplicant", "wpa_cli", "dhclient",
 +                    "ifplugd", "dhcdbd", "dhcpcd", "udhcpc",
 +                    "avahi-autoipd", "avahi-daemon", "wlassistant",
 +                    "wifibox"]
```

**Comment by squashen** [comment]
removed those processes and it starts automatic again^^

**Comment by anakin1028** [comment]
ok, I think we still need to kill these processes since users may just use `wpa_supplicant` and `dhclient` to connect to internet instead of using NetworkManager.

**Comment by Torkish** [comment]
Hi guys and sorry for my bad english
After launching wifiphisher, network manager stopped. I tried to restart it but it failed, i wait much time but still nothing.Even after reboot, network manager does not launch
I tried everything you said in that conversation. I reinstalled kali many times because of this problem. Can you help me please? Thank you

**Comment by anakin1028** [comment]
Have you tried the command `nmcli n on`?

**Comment by Torkish** [comment]
yes i did, it gives me an error : can't create NMClient object : time delay exceed 
it's not the real error message, I translated from french ton english

**Comment by sophron** [comment]
@anakin1028 It seems that we have many errors with `NetworkManager`. Maybe it makes sense to revert back and simply kill it, like the rest of the processes. 

What do you say?

**Comment by anakin1028** [comment]
Yes. I also think about that. I will rebase and drop this commit.

**Comment by sophron** [comment]
It's fine if you just push a new commit with the change.

**Pull request merged**